cmd/oras/internal/fileref/unix.go

```diff
@@ -24,14 +24,49 @@ import (
 
 // Parse parses file reference on unix.
 func Parse(reference string, defaultMediaType string) (filePath, mediaType string, err error) {
-	i := strings.LastIndex(reference, ":")
-	if i < 0 {
-		filePath, mediaType = reference, defaultMediaType
+	i := len(reference)
+	for {
+		// found the right most colon which is not escaped
+		i = strings.LastIndex(reference[:i], ":")
+		if i < 0 || !isEscaped(reference, i) {
+			break
+		}
+	}
+	if i < 0 || isEscaped(reference, i) {
+		filePath, mediaType = unescape(reference), defaultMediaType
 	} else {
-		filePath, mediaType = reference[:i], reference[i+1:]
+		filePath, mediaType = unescape(reference[:i]), reference[i+1:]
 	}
 	if filePath == "" {
 		return "", "", fmt.Errorf("found empty file path in %q", reference)
 	}
 	return filePath, mediaType, nil
 }
+
+func isEscaped(path string, offset int) bool {
+	cnt := 0
+	for i := offset - 1; i >= 0; i-- {
+		if path[i] != '\\' {
+			break
+		}
+	}
+	return cnt%2 != 0
```

Comment:
Isn't `cnt` always 0? Its value is not modified in the loop.

Reply:
Good catch! Added it with another unit test case.